### PR TITLE
UVOL 2.0

### PR DESCRIPTION
**Pull request by the-reality-engineer**

### CODE REVIEW

File: src/V2/player.ts [hunk 1]
```diff
@@ -0,0 +1,461 @@
...
+             */
+            this.onMeshBuffering(this.meshMap.size / this.geometryBufferSize)
+            return;
+        }
+
+
+        if (this.fileHeader.AudioURL) {
+            this.currentTime = this.audio.currentTime;
+        } else {
+            this.currentTime += this.clock.getDelta()
```
Comment (the-reality-engineer) [inline src/V2/player.ts]: @CITIZENDOT  We shouldn't use getDelta() here, we should just use Date.now() - playbackStartTime

Comment (CITIZENDOT) [reply src/V2/player.ts]: I was using [THREE.Clock](https://threejs.org/docs/#api/en/core/Clock) here, that uses perfomance.now internally when available. Shall I replace time handling with Date?

**Comment by CITIZENDOT** [comment]
@cla-bot check

**Comment by kungfooman** [comment]
Given the manifest:

```
{
    "Version": "v2",
    "DRCURLPattern": "LIAM_DRC/export_#####.obj.drc",
    "KTX2URLPattern": "LIAM_KTX2/export_#####.1k.ktx2",
    "AudioURL": "liamlow.mp3",
    "BatchSize": 5,
    "GeometryFrameCount": 250,
    "TextureSegmentCount": 50,
    "TotalFrames": 250,
    "GeometryFrameRate": 25,
    "TextureFrameRate": 25
}
```

Can we extract a bounding box from `drc` or `ktx2` without recalculating it by min/max-ing every vertex?

Otherwise we still don't have a AABB / bounding box for meshes, which is IMO the biggest performance/usability issue.

File: src/V2/player.ts [hunk 2]
```diff
@@ -79,6 +79,7 @@ export default class Player {
     private fileHeader: FileHeader | null
     private vertexShader: string
     private fragmentShader: string
+    //@ts-ignore
     private intervalId: NodeJS.Timer
```
Comment (kungfooman) [inline src/V2/player.ts]: Isn't that just a number? Why overcomplicate it twice?

Comment (CITIZENDOT) [reply src/V2/player.ts]: Yes. I'll make it a number.

**Comment by CITIZENDOT** [comment]
> Given the manifest:
> 
> ```
> {
>     "Version": "v2",
>     "DRCURLPattern": "LIAM_DRC/export_#####.obj.drc",
>     "KTX2URLPattern": "LIAM_KTX2/export_#####.1k.ktx2",
>     "AudioURL": "liamlow.mp3",
>     "BatchSize": 5,
>     "GeometryFrameCount": 250,
>     "TextureSegmentCount": 50,
>     "TotalFrames": 250,
>     "GeometryFrameRate": 25,
>     "TextureFrameRate": 25
> }
> ```
> 
> Can we extract a bounding box from `drc` or `ktx2` without recalculating it by min/max-ing every vertex?
> 
> Otherwise we still don't have a AABB / bounding box for meshes, which is IMO the biggest performance/usability issue.



I'm not sure if that's possible, since they're in compressed formats. Btw what are the advantages by precalculating it? When extracting the draco and ktx2 frames, `boundingSphere` is still calculated anyway, na? (please correct me If I'm wrong)

**Comment by kungfooman** [comment]
> I'm not sure if that's possible, since they're in compressed formats. Btw what are the advantages by precalculating it? When extracting the draco and ktx2 frames, `boundingSphere` is still calculated anyway, na? (please correct me If I'm wrong)

It's important to have aabb info as soon as possible, so you can properly focus an OrbitCamera e.g. onto it, while it's still loading (or not loaded at all, just the manifest - lets say in an editor where you layout your scene graph).

Other volumetric video formats give you the AABB straight away for that reason.

**Comment by CITIZENDOT** [comment]
> Given the manifest:
> 
> ```
> {
>     "Version": "v2",
>     "DRCURLPattern": "LIAM_DRC/export_#####.obj.drc",
>     "KTX2URLPattern": "LIAM_KTX2/export_#####.1k.ktx2",
>     "AudioURL": "liamlow.mp3",
>     "BatchSize": 5,
>     "GeometryFrameCount": 250,
>     "TextureSegmentCount": 50,
>     "TotalFrames": 250,
>     "GeometryFrameRate": 25,
>     "TextureFrameRate": 25
> }
> ```
> 
> Can we extract a bounding box from `drc` or `ktx2` without recalculating it by min/max-ing every vertex?
> 
> Otherwise we still don't have a AABB / bounding box for meshes, which is IMO the biggest performance/usability issue.

Thanks for this suggestion! It's work in progress now.

https://github.com/EtherealEngine/Universal-Volumetric/blob/25e08fc0013fcee6379c39665d4d48150faedafa/src/V2/player.ts#L488-L494

**Pull request merged**